ports/atmel-samd/boards/bradanlanestudio_coin_m0/mpconfigboard.mk

```diff
@@ -18,21 +18,39 @@ CHIP_FAMILY = samd21
 
 SPI_FLASH_FILESYSTEM = 1
 EXTERNAL_FLASH_DEVICES = "W25Q32JVxQ"
-LONGINT_IMPL = MPZ
+LONGINT_IMPL = NONE
 
 # the M0 Coin has limited functionality and many modules can be eliminated
 
 # there may be more modules which are of no used but will require further digging
 
+# Disable modules that are unusable on this special-purpose board.
+
+CIRCUITPY_FULL_BUILD = 0
+
+CIRCUITPY_AESIO = 0
+CIRCUITPY_AUDIOBUSIO = 0
+CIRCUITPY_AUDIOIO = 1
+CIRCUITPY_BITBANGIO = 0
+CIRCUITPY_BITMAPFILTER = 0
+CIRCUITPY_BITMAPTOOLS = 0
+CIRCUITPY_BLEIO = 0
+CIRCUITPY_BLEIO_HCI = 0
 CIRCUITPY_DISPLAYIO = 0
-CIRCUITPY_PULSEIO = 0
-CIRCUITPY_RTC = 0
+CIRCUITPY_FRAMEBUFFERIO = 0
+CIRCUITPY_FREQUENCYIO = 0
+CIRCUITPY_I2CTARGET = 0
+CIRCUITPY_KEYPAD = 0
 CIRCUITPY_ONEWIREIO = 0
+CIRCUITPY_PS2IO = 0
+CIRCUITPY_PULSEIO = 0
+CIRCUITPY_RGBMATRIX = 0
 CIRCUITPY_ROTARYIO = 0
+CIRCUITPY_RTC = 0
+CIRCUITPY_TOUCHIO = 1
+CIRCUITPY_USB_HID = 1
+CIRCUITPY_USB_MIDI = 0
```

Comment:
I think a lot of these are already turned off in `mpconfigboard.mk`, due to SAMD21. Circuit Playground Express is probably a good model.

Reply:
CPE does not turn off a lot of these. I looked at Big Honking Button as it too is a "special purpose" board.

Reply:
Well, certainly BLEIO, BLEIO_HCI, etc. are already off.

Reply:
`FOO ?=` means set to the given value if `FOO` is not already set (if it's undefined). `mpconfigboard.mk` is read first so that can do per-board settings, then `mpconfigport.mk`, then `py/circuitpy_mpconfig.mk`.

Reply:
OK. I can go through and attempt to remove redundant flags. I was _"just doing what others had done before me"_ :-)

Reply:
you don't have to get it perfect, but it's easier to read with fewer, and there are fewer implications of "well, this is off, but something not mentioned must be on", which isn't necessarily true.

Reply:
I'll do one "due diligence" pass and run my smokiest. I should have a new commit in about 30 minutes.

Reply:
thanks for the feedback!

Reply:
latest commit removed some of the redundant build flags (but added a couple too).